opensearch-operator/pkg/helpers/reconcile-helpers.go

```diff
@@ -112,12 +113,18 @@ func VersionCheck(instance *opsterv1.OpenSearchCluster) (int32, string) {
 		} else {
 			httpPort = 9200
 		}
-		securityconfigPath = "/usr/share/opensearch/config/opensearch-security"
+		securityConfigPort = httpPort
+		securityConfigPath = "/usr/share/opensearch/config/opensearch-security"
 	} else {
-		httpPort = 9300
-		securityconfigPath = "/usr/share/opensearch/plugins/opensearch-security/securityconfig"
+		if instance.Spec.General.HttpPort > 0 {
+			httpPort = instance.Spec.General.HttpPort
+		} else {
+			httpPort = 9200
+		}
```

Comment:
This entire if-else block is duplicated between the outer if and else blocks and can be moved outside.

Reply:
Good catch. Fixed.